pkg/utils/utils.go

```diff
@@ -189,17 +188,24 @@ func SplitYaml(gaintYAML []byte) ([][]byte, error) {
 
 // UpdateSideCarApply -
 func UpdateSideCarApply(sideCars []csmv1.ContainerTemplate, c *acorev1.ContainerApplyConfiguration) {
-	for _, side := range sideCars {
-		if *c.Name == side.Name {
-			if side.Image != "" {
-				*c.Image = string(side.Image)
+	UpdateContainerApply(sideCars, c)
+}
+
+// TODO: It seems like this function only acts on envs that are shared between
```

Comment:
Is this TODO needed?

Reply:
Mentioned in https://github.com/dell/csm-operator/pull/785#discussion_r1847147990 .